### PR TITLE
[VAULT-2774][VAULT-2775] Add sign and verify to VaultModule

### PR DESCRIPTION
**Pull request by sharifelgamal**
## Description of changes in release / Impact of release:
(insert text here)

## Documentation
(insert text here)

### Is this a breaking change?
- [ ] Yes
- [ ] No

### If you answered Yes then describe why is it so
(insert text here if applicable)

### Is there a way to disable the change?
- [ ] Use previous release
- [ ] Use a feature flag
- [ ] No

#### Additional details go here
(insert text here if applicable)

### CODE REVIEW

File: larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java [hunk 1]
```diff
@@ -188,17 +194,90 @@ public void delete(Object value, Object storage) throws EvalException {
         vault.delete(value, storage);
     }
 
+    @StarlarkMethod(
+        name = "sign",
+        doc = "signs a payload with given keyId",
+        parameters = {
+            @Param(
+                name = "keyId",
```
Comment (mjallday) [inline larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java:202]: should this be called `keyArn`? i think id and arn are distinct in aws platform.

https://docs.aws.amazon.com/kms/latest/developerguide/find-cmk-id-arn.html

File: larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java [hunk 2]
```diff
@@ -188,17 +194,90 @@ public void delete(Object value, Object storage) throws EvalException {
...
+            @Param(
+                name = "message",
+                doc = "the payload to sign",
+                allowedTypes = {
+                    @ParamType(type = String.class)
+                }
+            ),
+            @Param(
+                name = "algorithm",
+                doc = "the algorith to use to sign.",
```
Comment (mjallday) [inline larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java:217]: do we need to include the algorithms here or make this an enum? i cannot guess what values go here so i need docs.

Comment (sharifelgamal) [reply larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java:217]: i'm fine adding the choices in the doc.

File: larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java (hunk 1 above)
Comment (mahmoudimus) [reply larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java:202]: Are we exposing AWS internals too much? I understand the lack of abstraction, but would this be a breaking change if someone decided to use a different key provider?

File: larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java (hunk 2 above)
Comment (mahmoudimus) [inline larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java:217]: spelling, should be algorithm (fyi).

File: larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java [hunk 3]
```diff
@@ -214,25 +214,26 @@ public void delete(Object value, Object storage) throws EvalException {
             ),
             @Param(
                 name = "algorithm",
-                doc = "the algorith to use to sign.",
+                doc = "the algorithm to use to sign, valid choices are: RSASSA_PSS_SHA_256, RSASSA_PSS_SHA_384, " +
```
Comment (mjallday) [inline larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java:217]: 👍

**Comment by mjallday** [comment]
do we have to pass [MessageType](https://docs.aws.amazon.com/kms/latest/APIReference/API_Sign.html#API_Sign_RequestSyntax)?

**Comment by sharifelgamal** [comment]
> do we have to pass [MessageType](https://docs.aws.amazon.com/kms/latest/APIReference/API_Sign.html#API_Sign_RequestSyntax)?

We're assuming MessageType.RAW for everyone

File: larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java (hunk 1 above)
Comment (sharifelgamal) [reply larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java]: we can keep it as keyId to keep it as generic as possible, but if we end up supporting HSMs outside KMS, things will get really complicated really fast

Comment (mjallday) [reply larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java]: yeah, we can add a `provider` parameter later. i see your point on `keyId`.

**Review by mahmoudimus — APPROVED** [review]

File: larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java (hunk 1 above)
Comment (Iapetus999) [reply larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java:202]: Every place we use this in the current code base, we use `...KEY_ID`

**Comment by sharifelgamal** [comment]
Moved to https://github.com/verygoodsecurity/starlarky/pull/423

**Pull request closed without merging**